operator/pkg/controllers/utils.go

```diff
@@ -586,3 +586,86 @@ func applyRateLimitToContainerArg(container map[string]interface{}, rateLimit *c
 	}
 	return nil
 }
+
+func ApplyContainerPprofConfig(m *manifest.Objects, targetControllerName string, pprofConfig *customizev1alpha1.PprofConfig) error {
+	if pprofConfig == nil {
+		return nil
+	}
+
+	var (
+		targetContainerName string
+		targetControllerGVK schema.GroupVersionKind
+	)
+	switch targetControllerName {
+	case "cnrm-controller-manager":
+		targetContainerName = "manager"
+		targetControllerGVK = schema.GroupVersionKind{
+			Group:   appsv1.SchemeGroupVersion.Group,
+			Version: appsv1.SchemeGroupVersion.Version,
+			Kind:    "StatefulSet",
+		}
+	default:
+		return fmt.Errorf("pprof config customization for %s is not supported. "+
+			"Supported controllers: %s",
+			targetControllerName, strings.Join(customizev1alpha1.SupportedPprofControllers, ", "))
+	}
+
+	for _, item := range m.Items {
+		if item.GroupVersionKind() != targetControllerGVK {
+			continue
+		}
+		if !strings.HasPrefix(item.GetName(), targetControllerName) {
+			continue
+		}
+		if err := item.MutateContainers(customizePprofConfigFn(targetContainerName, pprofConfig)); err != nil {
+			return err
+		}
+		break // we already found the matching controller, no need to keep looking.
```

Comment:
My 2c: the small savings in CPU are not worth breaking.  I think it's better to return nil here and making it an error if we don't find a match (i.e. return error at the end of the function)

Another option is to count the number of containers changed and return an error if there was not exactly one changed.

Reply:
Going with 2 but also changing ApplyContainerRateLimit to match.